marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx

```diff
@@ -259,6 +266,120 @@ describe('RTL Source-to-entity map tests', () => {
     });
 });
 
+
+test('Nested source data - Right XPATH expression',  async() => {
+    axiosMock.post.mockImplementationOnce(data.mapProps.updateMappingArtifact);
+    const { getByText,getAllByText, queryByText, getAllByTestId, getAllByRole, debug} = render(<SourceToEntityMap {...data.mapProps}  mappingVisible={true}/>);
+    expect(getByText('Source Data')).toBeInTheDocument();
+    expect(getByText('Entity: Person')).toBeInTheDocument();
+    expect(getByText('Test')).toBeEnabled();
+
+    //Ensure source selectors are present
+    let sourceSelectors = getAllByTestId("listSources");
+
+    expect(sourceSelectors.length).toEqual(8)
+
+    //corresponds to 'itemTypes' source selector
+    fireEvent.click(sourceSelectors[3]);
+
+    await(waitForElement(() =>  getAllByRole("option"),{"timeout":200}))
+    let firstName = getAllByText("FirstNamePreferred");
+    expect(firstName.length).toEqual(2)
+
+    //Check if indentation is right
+    expect(firstName[1]).toHaveStyle("line-height: 2vh; text-indent: 20px;");
```

Comment:
Since we are able to verify indentation using RTL, can we add a test using XML data to verify indentation, namespaces and attributes.

Reply:
Added test for XML data